### PR TITLE
Partio::cloneSchema() implementation and build maintenance

### PR DESCRIPTION
**Pull request by davvid**
Add an API method for cloning a particle's schema.

### CODE REVIEW

File: src/lib/core/Particle.cpp [hunk 1]
```diff
@@ -68,6 +68,20 @@ createInterleave()
 }
 
 
+ParticlesDataMutable*
+clone(const ParticlesData& other)
```
Comment (aselle) [inline src/lib/core/Particle.cpp]: maybe call cloneSchema

File: src/lib/core/Particle.cpp [hunk 2]
```diff
@@ -68,6 +68,20 @@ createInterleave()
...
+{
+    ParticlesDataMutable* p = create();
+
+    ParticleAttribute attr;
```
Comment (aselle) [inline src/lib/core/Particle.cpp]: detail attributes also need cloning

File: src/tests/testclonecopy.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
+/*
```
Comment (aselle) [inline src/tests/testclonecopy.cpp]: Nice, a test!

**Comment by davvid** [comment]
@aselle I renamed `clone()` to `cloneSchema()` and added support for fixed (detail) attributes.

**Pull request merged**